articles/build-an-outlook-clone-using-react-hooks/index.md

```diff
@@ -0,0 +1,881 @@
+---
+layout: engineering-education
+status: publish
+published: true
+slug: build-an-outlook-clone-using-react-hooks
+title: React Hooks and Building Outlook Clone
+description: Complete guide where we discuss react hooks and have written optimized code following up on our earlier topic of how to build an outlook clone using React.
+author: lalithnaryan-c
+date: 2020-07-23T00:00:00-07:00
+topics: []
+excerpt_separator: <!--more-->
+images:
+
+  - url: /engineering-education/build-an-outlook-clone-using-react-hooks/hero.jpg
+    alt: outlook clone example image
+---
+Welcome to Hook part of the React series. This time we cover implementing the same outlook clone that we built earlier. But this time, we use React Hooks. The earlier codebase was repetitive and there is a scope for optimization of the entire project. To sum up the earlier attempt at building an outlook clone, we made API calls and displayed emails according to the filter requested. We also implemented local storage.
+
+<!--more-->
+
+Let's think about this for a minute. When we implemented local storage, was it a good idea to store the entire state? This is a factor that needs some consideration of the user interface. Upon reloading, we can simply show the user the set of reading and favorite mails and refresh the layout to the default option. This is the ideal approach to building projects. First, we build them and then work with various optimization in terms of performance, user experience, and code quality.
```

Comment:
> This is a factor that needs some consideration of the user interface.
This should be refactored to flow better. Something along the lines of, "While this made sense during the last project, it does not produce the best experience for the user."

> This is the ideal approach to building projects. First, we build them and then work with various optimization in terms of performance, user experience, and code quality.
This statement is confusing, consider revising to make it clear that you are changing subjects and are now clarifying why you are refactoring a working product.

Reply:
I will explain these in detail and with much more clarity. Thank you for the feedback